tracer/src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs

```diff
@@ -39,36 +39,43 @@ static Startup()
+            try
             {
-                StartupLogger.Log("Managed profiler directory doesn't exist. Automatic instrumentation will be disabled");
-                return;
-            }
+                ManagedProfilerDirectory = ResolveManagedProfilerDirectory();
+                if (ManagedProfilerDirectory is null)
+                {
+                    StartupLogger.Log("Managed profiler directory doesn't exist. Automatic instrumentation will be disabled");
+                    return;
+                }
 
-            StartupLogger.Debug("Resolving managed profiler directory to: {0}", ManagedProfilerDirectory);
+                StartupLogger.Debug("Resolving managed profiler directory to: {0}", ManagedProfilerDirectory);
 
-            try
-            {
-                AppDomain.CurrentDomain.AssemblyResolve += AssemblyResolve_ManagedProfilerDependencies;
-            }
-            catch (Exception ex)
-            {
-                StartupLogger.Log(ex, "Unable to register a callback to the CurrentDomain.AssemblyResolve event.");
-            }
+                try
+                {
+                    AppDomain.CurrentDomain.AssemblyResolve += AssemblyResolve_ManagedProfilerDependencies;
+                }
+                catch (Exception ex)
+                {
+                    StartupLogger.Log(ex, "Unable to register a callback to the CurrentDomain.AssemblyResolve event.");
+                }
 
-            var runInAas = ReadBooleanEnvironmentVariable(AzureAppServicesKey, false);
-            if (runInAas)
+                var runInAas = ReadBooleanEnvironmentVariable(AzureAppServicesKey, false);
+                if (runInAas)
+                {
+                    // With V3, pretty much all scenarios require the trace-agent and dogstatsd, so we enable them by default
+                    StartupLogger.Log("Invoking managed method to start external processes.");
+                    TryInvokeManagedMethod("Datadog.Trace.AgentProcessManager", "Initialize", "Datadog.Trace.AgentProcessManagerLoader");
+                }
+
+                // We need to invoke the managed tracer regardless of whether tracing is enabled
+                // because other products rely on it
+                StartupLogger.Log("Invoking managed tracer.");
+                TryInvokeManagedMethod("Datadog.Trace.ClrProfiler.Instrumentation", "Initialize", "Datadog.Trace.ClrProfiler.InstrumentationLoader");
+            }
+            catch
             {
-                // With V3, pretty much all scenarios require the trace-agent and dogstatsd, so we enable them by default
-                StartupLogger.Log("Invoking managed method to start external processes.");
-                TryInvokeManagedMethod("Datadog.Trace.AgentProcessManager", "Initialize", "Datadog.Trace.AgentProcessManagerLoader");
+                // Swallowing any errors here, as something went _very_ wrong, so trying to log might cause issues itself
```

Comment:
I'm a bit worried about this one. We don't log and we don't crash, how are we going to detect those 🤔 

Maybe we should swallow it only in SSI? Or break the golden rule and log something to stderr?

Reply:
I mean, short answer we wouldn't 😅 

We could _try_ to log using `StartupLogger.Log` and swallow _that_ if it fails, that's maybe a good first step.

> Maybe we should swallow it only in SSI

Yeah, that's a possibility I'm ok with too 🤔 I'll implement both here.

> log something to stderr

Nah 😛

Reply:
> > log something to stderr
>
> Nah 😛

Why not?

Reply:
Writing to stdout and stderr can break logging mechanisms if they expect logs to be in a specific format. Pretty much as a matter of policy, we don't write to these. Given we're trying to be "as safe as possible" here, writing to stderr doesn't make a whole lot of sense